pwndbg/dbg/lldb/__init__.py

```diff
@@ -586,6 +588,66 @@ def set_enabled(self, enabled: bool) -> None:
         self.inner.SetEnabled(enabled)
 
 
+class OneShotAwaitable:
+    """
+    Used as part of the logic for the execution controller. This is an Awaitable
+    object that yields the value passed to its constructor exactly once.
+    """
+
+    def __init__(self, value: Any):
+        self.value = value
+
+    def __await__(self) -> Generator[Any, Any, None]:
+        yield self.value
+
+
+class YieldContinue:
+    """
+    Continues execution of the process until the breakpoint or watchpoint given
+    in the constructor is hit or the operation is cancelled.
+
+    This class is part of the execution controller system, so it is intented to
+    be yielded by the async function with access to an execution controller, and
+    caught and hanlded by the event loop in the LLDB Pwndbg CLI.
+    """
+
+    target: LLDBStopPoint
+
+    def __init__(self, target: LLDBStopPoint):
+        self.target = target
+
+
+class YieldSingleStep:
+    """
+    Moves execution of the process being debugged forward by one instruction.
+
+    This class is part of the execution controller system, so it is intented to
+    be yielded by the async function with access to an execution controller, and
+    caught and hanlded by the event loop in the LLDB Pwndbg CLI.
+    """
+
+    pass
```

Comment:
Not really, we just yield them up to the event loop in the Pwndbg CLI, which then distinguishes them by type, and handles them. The way we do it, having two classes is hardly any different from having a single class, since we'd still need a way to distinguish between the continue and single step cases.